tools/svelte-actions/src/ripple.ts

```diff
@@ -24,14 +32,14 @@ class Ripple {
   private element: HTMLElement;
   private triggerElement: HTMLElement;
   private options: RippleOptions;
-  private animationElement: HTMLElement;
+  private animationElement: HTMLElement | undefined = undefined;
 
   constructor(element: HTMLElement, options: RippleOptions) {
     this.element = element;
     this.options = { ...RippleDefaults, ...options };
     this.injectContainerStyle();
     if (this.options.triggerOnParent) {
-      this.triggerElement = this.element.parentElement;
+      this.triggerElement = this.element.parentElement || this.element;
```

Comment:
Where does this occur? Seems a little counter-intuitive if someone uses `triggerOnParent` and it still triggers on the element. Maybe we could warn or just error or `if` there?

Reply:
this should only happen if an element has no parent. which is ... `body` ? or even just `html` ?
the main reason for adding it was making typescript happy :P